### PR TITLE
add zero migration blog and images

### PR DESCRIPTION
**Pull request by mbroz2**
Post for https://github.com/OpenLiberty/blogs/issues/1671

https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/2024/09/30/zero-migration.html

image

### CODE REVIEW

**Review by gljrobinson — DISMISSED** [review]
Looks great to me, thanks Michal

**Comment by mbroz2** [comment]
@lauracowen @GraceJansen the blog is currently dated for 9/30.  What date would you like for the publication?

File: posts/2024-09-30-zero-migration.adoc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+open-graph-image-alt: Open Liberty Logo
+---
+= Effortless updates with Liberty\'s zero migration
+Michal Broz <https://github.com/mbroz2>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+In today's production environments it is not only expected, but critical, to keep your technology stack as up-to-date as possible. This helps to mitigate security risks, while also benefiting from bug fixes, performance improvements, and new features and function that can help drive your business's success. This is true whether you're deploying to a cloud or locally, and applies regardless of the deployment type, whether container images, VMs, or bare metal.
```
Comment (dmuelle) [inline posts/2024-09-30-zero-migration.adoc]:
```suggestion
In today's production environments, it is not only expected, but critical to keep your technology stack as up to date as possible. Staying current helps to mitigate security risks, while also benefiting from bug fixes, performance improvements, and new features and function that can help drive your business's success. This is true whether you're deploying to a cloud or locally, and applies regardless of the deployment type, whether container images, VMs, or bare metal.
```

File: posts/2024-09-30-zero-migration.adoc [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+## The challenge of keeping microservices up-to-date
```
Comment (dmuelle) [inline posts/2024-09-30-zero-migration.adoc]:
```suggestion
## The challenge of keeping microservices up to date
```

weird rule, but you generally only hyphenate compound adjectives when the precede the noun they modify, not when they follow it, as here.

File: posts/2024-09-30-zero-migration.adoc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+Now multiply these concerns by the hundreds or thousands of deployments of your microservices, each containing an instance of the application runtime, and it's no wonder that many applications remain on old versions and teams pursue upgrades only when deemed absolutely necessary. This growing technical debt not only delays the benefits provided by newer releases, including security fixes, but can also snowball into making the eventual migration process that much more difficult and risky.
+
+Although container image scanning is improving, containers can further exasperate the problem by obfuscating old, possibly unsupported, runtimes and dependencies. Teams might not even be aware of an aging runtime and the potential security risks it could pose.
+
+## Liberty's zero-migration architecture
+
+This is where the benefit of Liberty's zero-migration architecture comes in. Zero migration is a core design principal of the Liberty application runtime which seeks to make updating Liberty quick and easy.
```
Comment (dmuelle) [inline posts/2024-09-30-zero-migration.adoc]:
```suggestion
This is where the benefit of Liberty's zero-migration architecture comes in. Zero migration is a core design principal of the Liberty application runtime that seeks to make updating Liberty quick and easy.
```

File: posts/2024-09-30-zero-migration.adoc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+### Liberty features and versions
+
+Liberty features are what provide Liberty with its modularity. Each Liberty capability is associated with a feature; for example, the Admin Center feature provides an administrative GUI and the Audit feature provides tracking of auditable events. Features are composed of the feature name and the feature version, which you specify in the link:/docs/latest/reference/config/server-configuration-overview.html[configuration files] (e.g. `server.xml` file) for the Liberty instance on which the application runs. The versions of the features are not tied to the version of Liberty.
+
+You can upgrade the version of Liberty on which your application runs (e.g. to get a security update) without upgrading the versions of the features that the application uses. Whenever a breaking change needs to be introduced into a feature, its feature version is incremented, but Liberty continues to support all previous versions as well as the new version. This means that you're in control of if and when you want your application to use a newer version of the feature.
```
Comment (dmuelle) [inline posts/2024-09-30-zero-migration.adoc]:
```suggestion
You can upgrade the version of Liberty on which your application runs (e.g. to get a security update) without upgrading the versions of the features that the application uses. Whenever a breaking change needs to be introduced into a feature, its feature version is incremented, but Liberty continues to support all previous versions as well as the new version. This means that you control if and when you want your application to use a newer version of the feature.
```

File: posts/2024-09-30-zero-migration.adoc [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+For example, if your application is using MicroProfile Config 2.0 (the `mpConfig-2.0` feature), it's up to you if and when you want to move up to `mpConfig-3.0`, `mpConfig-3.1`, or any other future versions of MicroProfile Config. Your application can continue to use `mpConfig-2.0` while Liberty receives updates, including bug and security fixes, new features and functionality, and performance enhancements.
+
+## Upgrading to a new version of Liberty
```
Comment (dmuelle) [inline posts/2024-09-30-zero-migration.adoc]:
```suggestion
## Updating to a new version of Liberty
```

File: posts/2024-09-30-zero-migration.adoc [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+// [.img_border_light]
+// image::img/blog/zero_migration-timeline_and_table.png[Release timeline and table,width=100%,align="center"]
+
+// [.img_border_light]
+// image::img/blog/zero_migration-timeline.png[Release timeline,width=100%,align="center"]
+
+.Liberty Support Lifecycle
+[cols="3,2,2", options="header"]
+|=======================================================================================
+|                          | All releases | Releases ending in “.3”, “.6”, “.9” or “.12”
```
Comment (dmuelle) [inline posts/2024-09-30-zero-migration.adoc]:
```suggestion
| Support status                       | All releases | Releases ending in “.3”, “.6”, “.9” or “.12”
```

not sure this is the best heading but the table will fail accessibility with an empty header cell

File: posts/2024-09-30-zero-migration.adoc [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+Our prebuilt link:/docs/latest/container-images.html[Liberty container images] are rebuilt weekly for the versions eligible for the proactive security fixes, pulling in any Java and UBI fixes that have been released since the previous build of the image. This provides you with a great base image for your application, while retaining control over which Liberty iFixes you want to apply to your image.
+
+## Caveats
+
+It wouldn't be the full story without the caveats. Thankfully, when it comes to zero migration, there are only a few and they mainly relate to important security fixes and circumstances outside of the control of the Liberty developers:
+
+#### Exceptions:
+
+- Security fixes: Whenever performing a security hardening or patching, we try to maintain the existing behavior of Liberty, configuration, or features. When this is not possible, we work to limit the scope of the change to only what is needed to address the security concern.
```
Comment (dmuelle) [inline posts/2024-09-30-zero-migration.adoc]:
```suggestion
- Security fixes: Whenever we perform a security hardening or patching, we try to maintain the existing behavior of Liberty, configuration, or features. When this is not possible, we work to limit the scope of the change to only what is needed to address the security concern.
```

File: posts/2024-09-30-zero-migration.adoc [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+// [.img_border_light]
+// image::img/blog/zero_migration-iFixes.png[Releases with iFixes,width=100%,align="center"]
+
+[.img_border_light]
+image::img/blog/zero_migration-iFixes_includes.png[Releases with iFixes,width=100%,align="center"]
+
+## Zero migration vs technical debt
+
+Zero migration not only reduces your technical debt, the overhead, of keeping a runtime current, but in many cases eliminates it entirely, allowing you to focus on the higher value items of development and operations.
```
Comment (dmuelle) [inline posts/2024-09-30-zero-migration.adoc]:
```suggestion
Zero migration not only reduces your technical debt, the overhead of keeping a runtime current, but in many cases eliminates it entirely, allowing you to focus on the higher value items of development and operations.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
a few minor  suggestions- can we target after the beta blog publishes on tuesday? The tags file has some changes that have yet to make it to prod.

**Review by dmuelle — APPROVED** [review]

**Pull request merged**